From a7c7e96e17937ea0780261101a4c45383df71134 Mon Sep 17 00:00:00 2001
From: Werner Hoch <werner.ho@gmx.de>
Date: Sun, 18 Oct 2009 23:03:23 +0100
Subject: [PATCH] gschem: Fix rendering with zoomed out dashed lines. Bug #2876373

Rather than using SCREENabs, use the screen_width function to
determine the on-screen size of dash elements. This clamps the
width at a rendereable size, so cairo does not get tripped into
an error state (and stop drawing other objects on the schematic).
---
 gschem/src/gschem_cairo.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gschem/src/gschem_cairo.c b/gschem/src/gschem_cairo.c
index e9a8e5c..2e7882d 100644
--- a/gschem/src/gschem_cairo.c
+++ b/gschem/src/gschem_cairo.c
@@ -292,8 +292,8 @@ void gschem_cairo_stroke (GSCHEM_TOPLEVEL *w_current, int line_type, int line_en
   int width, length, space;
 
   width  = screen_width (w_current, wwidth);
-  length = SCREENabs (w_current, wlength);
-  space  = SCREENabs (w_current, wspace);
+  length = screen_width (w_current, wlength);
+  space  = screen_width (w_current, wspace);
   offset = ((width % 2) == 0) ? 0 : 0.5;
 
   cairo_set_line_width (w_current->cr, width);
-- 
1.6.5

